### PR TITLE
Check db migrations run before hubs start on install

### LINKED ISSUE #704: Make sure db migrations have been applied when the hub runs in Helm CI tests
Thanks @makkes for pointing this out!

### PR DESCRIPTION
**Pull request by tegioz**
Fixes #704

Signed-off-by: Sergio Castaño Arteaga <tegioz@icloud.com>

### CODE REVIEW

File: charts/artifact-hub/templates/hub_deployment.yaml [hunk 1]
```diff
@@ -26,6 +25,9 @@ spec:
       imagePullSecrets:
         {{- toYaml . | nindent 8 }}
     {{- end }}
+      {{- if .Release.IsInstall }}
+      serviceAccountName: hub
+      {{- end }}
```
Comment (makkes) [inline charts/artifact-hub/templates/hub_deployment.yaml:30]: I'm not sure I understand why this SA should only be used when installing.

**Review by makkes — APPROVED** [review]
lgtm, just one question/remark.

File: charts/artifact-hub/templates/hub_deployment.yaml (hunk 1 above)
Comment (tegioz) [reply charts/artifact-hub/templates/hub_deployment.yaml:30]: Only when installing we check if the db-migrator job has completed. After the first upgrade, the hub should never talk anymore to the k8s api, so it doesn't need the service account.

Comment (makkes) [reply charts/artifact-hub/templates/hub_deployment.yaml:30]: I understand.

Comment (tegioz) [reply charts/artifact-hub/templates/hub_deployment.yaml:30]: Thanks for your help @makkes. I'll merge this now then. Happy to reopen the discussion and make the necessary changes if you can think of any issue at some point.

**Pull request merged**